x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/parser/ExpressionBuilder.java

```diff
@@ -6,6 +6,188 @@
+import org.elasticsearch.xpack.ql.expression.predicate.operator.arithmetic.Neg;
+import org.elasticsearch.xpack.ql.expression.predicate.operator.arithmetic.Sub;
+import org.elasticsearch.xpack.ql.expression.predicate.operator.comparison.Equals;
+import org.elasticsearch.xpack.ql.expression.predicate.operator.comparison.GreaterThan;
+import org.elasticsearch.xpack.ql.expression.predicate.operator.comparison.GreaterThanOrEqual;
+import org.elasticsearch.xpack.ql.expression.predicate.operator.comparison.LessThan;
+import org.elasticsearch.xpack.ql.expression.predicate.operator.comparison.LessThanOrEqual;
+import org.elasticsearch.xpack.ql.expression.predicate.operator.comparison.NotEquals;
+import org.elasticsearch.xpack.ql.tree.Source;
 
+import java.util.Arrays;
+import java.util.List;
+
+
+public class ExpressionBuilder extends LiteralBuilder {
+
+    protected Expression expression(ParseTree ctx) {
+        return typedParsing(ctx, Expression.class);
+    }
+
+    protected List<Expression> expressions(List<? extends ParserRuleContext> contexts) {
+        return visitList(contexts, Expression.class);
+    }
+
+    @Override
+    public Expression visitSingleExpression(EqlBaseParser.SingleExpressionContext ctx) {
+        return expression(ctx.expression());
+    }
+
+    @Override
+    public Expression visitArithmeticUnary(ArithmeticUnaryContext ctx) {
+        Expression expr = expression(ctx.valueExpression());
+        Source source = source(ctx);
+        int type = ctx.operator.getType();
+
+        return type == EqlBaseParser.MINUS ? new Neg(source, expr) : expr;
+    }
+
+    @Override
+    public Expression visitArithmeticBinary(EqlBaseParser.ArithmeticBinaryContext ctx) {
+        Expression left = expression(ctx.left);
+        Expression right = expression(ctx.right);
+        Source source = source(ctx);
+        int type = ctx.operator.getType();
+
+        switch (type) {
+            case EqlBaseParser.ASTERISK:
+                return new Mul(source, left, right);
+            case EqlBaseParser.SLASH:
+                return new Div(source, left, right);
+            case EqlBaseParser.PERCENT:
+                return new Mod(source, left, right);
+            case EqlBaseParser.PLUS:
+                return new Add(source, left, right);
+            case EqlBaseParser.MINUS:
+                return new Sub(source, left, right);
+            default:
+                throw new ParsingException(source, "Unknown arithmetic {}", source.text());
+        }
+    }
+
+    @Override
+    public Expression visitComparison(ComparisonContext ctx) {
+        Expression left = expression(ctx.left);
+        Expression right = expression(ctx.right);
+        TerminalNode op = (TerminalNode) ctx.comparisonOperator().getChild(0);
+
+        Source source = source(ctx);
+
+        // check if the RHS is a wildcard string and convert to a function check instead
+        if (right instanceof Literal) {
```

Comment:
Good catch. Yeah that makes sense to move this.
This will catch the syntax `command_line == "*some*wildcard* expression*"`. Even more broadly, any time the RHS of `!=` or `==` is a string with an asterisk.

Reply:
removed this handling in eba19f752782f8780025eaecafbe3946722f337f.
will add back later when an analyzer is created

Reply:
> This will catch the syntax `command_line == "*some*wildcard* expression*"`. Even more broadly, any time the RHS of `!=` or `==` is a string with an asterisk.

This seems similar to the `Regex` functions in QL that map to `RLIKE` and `LIKE` in SQL. Essentially it's a scalar around the regex pattern by encapsulating its logic into a dedicated class.
See https://github.com/elastic/elasticsearch/blob/789724ac2cc726de603849b4eeb8194da7528bcc/x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/expression/predicate/regex/LikePattern.java and https://github.com/elastic/elasticsearch/blob/789724ac2cc726de603849b4eeb8194da7528bcc/x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/expression/predicate/regex/RegexMatch.java hierarchy.

If you know for sure in the parser that `!=` and `==` followed by a string with asterisk is wildcard match, you can instantiate the function directly:
https://github.com/elastic/elasticsearch/blob/789724ac2cc726de603849b4eeb8194da7528bcc/x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/parser/ExpressionBuilder.java#L223

This has the nice side-effect of using a function that does not have to be declared in the registry (and thus cannot be instantiated by a query).